### PR TITLE
Document HNC labels and annotations

### LINKED ISSUE #1301: Document HNCs usage of annotations and labels
This includes which annotations/labels HNC uses, creates, or alters for both original and propagated resources. 

This should likely live here: https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/docs/user-guide/concepts.md

### PR DESCRIPTION
**Pull request by adrianludwin**
Published at https://github.com/adrianludwin/multi-tenancy/blob/doc-annot/incubator/hnc/docs/user-guide/concepts.md#admin-labels.

Fixes #1301

### CODE REVIEW

File: incubator/hnc/docs/user-guide/concepts.md [hunk 1]
```diff
@@ -455,5 +456,108 @@ problem, and will generally require human intervention to resolve.
...
+on any source object, such as objects created by Helm or Config Sync.
+
+This label has no meaning _to_ HNC; it's only provided as a way for users to
+determine that HNC created an object.
+
+See also `hnc.x-k8s.io/inherited-from`.
+
+#### hnc.x-k8s.io/inherited-from
+
+HNC sets this label on any object that is propagates, similar to
```
Comment (GinnyJI) [inline incubator/hnc/docs/user-guide/concepts.md:483]: Do you mean "it propagates"?

File: incubator/hnc/docs/user-guide/concepts.md [hunk 2]
```diff
@@ -455,5 +456,108 @@ problem, and will generally require human intervention to resolve.
...
+management of the namespaces.
+
+Config Sync and ACM both have their own concept of namespace hierarchy, which
+predates HNC's (HNC is actually based on these products). Unlike HNC, these
+products only instantiate the _leaf_ namespaces on a cluster, with all
+higher-level namespace (which they call _abstract namespaces_) only existing on
+a filesystem in a Git repo. However, they have adopted HNC's [tree
+labels](#basic-labels) to allow the leaf namespaces to be selected by subtree.
+
+Ordinarily, HNC overwrites any tree labels with its own, but by setting the
```
Comment (GinnyJI) [inline incubator/hnc/docs/user-guide/concepts.md:506]: "any tree label"?

File: incubator/hnc/docs/user-guide/concepts.md [hunk 3]
```diff
@@ -455,5 +456,108 @@ problem, and will generally require human intervention to resolve.
...
+Generally speaking, you should never have to look at or modify this annotation.
+The one exception is if you would like to convert a subnamespace into a full
+namespace - which is to say, you no longer want its lifetime to be controlled by
+its anchor. In such cases, you can do the following:
+
+1. Remove the `hnc.x-k8s.io/subnamespace-of` annotation from the subnamespace.
+2. Ensure that the anchor in the parent is in the `Conflict` state, and then
+   delete the anchor.
+
+At this point, the namepsace will still be a _child_ of the same namespace, but
```
Comment (GinnyJI) [inline incubator/hnc/docs/user-guide/concepts.md:548]: typo: namespace

File: incubator/hnc/docs/user-guide/concepts.md [hunk 4]
```diff
@@ -455,5 +456,108 @@ problem, and will generally require human intervention to resolve.
...
+<a name="admin-labels">
+
+### Labels and annotations
+
+HNC both adds and respects certain annotations and labels on namespaces and
+objects. Typically, most users (or admins) will never need to care about these,
+but occasionally they may cause some odd changes in behaviour that you need to
+be aware of.
+
+#### app.kubernetes.io/managed-by
```
Comment (yiqigao217) [inline incubator/hnc/docs/user-guide/concepts.md:470]: Maybe indicate if it's label or annotation in the each sub-header?

File: incubator/hnc/docs/user-guide/concepts.md [hunk 5]
```diff
@@ -455,5 +456,108 @@ problem, and will generally require human intervention to resolve.
 
 > _Note: HNC v0.5 reported issues with objects as part of the non-standard
 > condition schema. These have been removed and replaced by standard Events in
-> HNC v0.6 since events are more standard, scalable and loggable.
+> HNC v0.6 since events are more standard, scalable and loggable._
+
+<a name="admin-labels">
+
+### Labels and annotations
```
Comment (yiqigao217) [inline incubator/hnc/docs/user-guide/concepts.md:463]: Maybe also categorize the labels/annotations into two:
1) Added by HNC (most of them)
2) Added by users (exceptions)

File: incubator/hnc/docs/user-guide/concepts.md (hunk 1 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/concepts.md:483]: Done

File: incubator/hnc/docs/user-guide/concepts.md (hunk 2 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/concepts.md:506]: Changed to "any existing tree label."

File: incubator/hnc/docs/user-guide/concepts.md (hunk 3 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/concepts.md:548]: Done

File: incubator/hnc/docs/user-guide/concepts.md (hunk 4 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/concepts.md:470]: Done

File: incubator/hnc/docs/user-guide/concepts.md (hunk 5 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/concepts.md:463]: Done

**Review by yiqigao217 — COMMENTED** [review]
/lgtm
/hold

Just holding if @GinnyJI wants to have another eye. Please feel free to remove it.

**Comment by rjbez17** [comment]
Sorry, started this earlier and didn't get back until now but it lgtm.

**Pull request merged**